### PR TITLE
Pydineof download changes

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses #(issue)

Pulls in bug fixes found in #82 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

 - Unit tests

**Test Configuration**:
* Operating system: MacOS
* Version number: Python 3.8
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Comment by rstoneback** [comment]
Tests are running into a documentation issue:
```
(examples/ex_init: line   11) broken    https://www.hao.ucar.edu/modeling/tgcm/tie.php - HTTPSConnectionPool(host='www.hao.ucar.edu', port=443): Max retries exceeded with url: /modeling/tgcm/tie.php (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f2e08559d60>: Failed to establish a new connection: [Errno 110] Connection timed out'))
```

**Comment by rstoneback** [comment]
Same issue over in #83 and thus not something caused here.

**Comment by aburrell** [comment]
> Tests are running into a documentation issue

This happens frequently, but the link works and I don't know a better one to include.

**Comment by aburrell** [comment]
Is this PR in lieu of #82?

File: pysatModels/models/pydineof_dineof.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 """
-Supports exported model data from pyDINEOF, a Python package
+Support exported model data from pyDINEOF, a Python package
 that interfaces with a version of Data Interpolation Empirical
 Orthogonal Functions (DINEOFs). This module couples into
 the systematic export pyDINEOF format and thus should support
```
Comment (aburrell) [inline pysatModels/models/pydineof_dineof.py:6]: This needs to be reduced to one line, the rest can be moved to the "Note" section.

File: pysatModels/models/pydineof_dineof.py [hunk 2]
```diff
@@ -76,7 +76,7 @@
 
 
 def init(self):
-    """Initializes the Instrument object with instrument specific values.
+    """Initialize the Instrument object with instrument specific values.
     """
```
Comment (aburrell) [inline pysatModels/models/pydineof_dineof.py:80]:
```suggestion
    """Initialize the Instrument object with instrument specific values."""
```

Here and elsewhere

**Comment by rstoneback** [comment]
Github won't let me use my password anymore and I'm having a variety of issues with my IDE and GitHub on using a token.  I can't push any updates at the moment.  It may take me a bit to sort this out.

**Comment by rstoneback** [comment]
> > Tests are running into a documentation issue
> 
> This happens frequently, but the link works and I don't know a better one to include.

Thanks for the note. Yeah, I checked the website manually on my end and it loaded fine. I think it is the site complaining.

**Comment by rstoneback** [comment]
> Is this PR in lieu of #82?

Yep. I wasn't able to get the windows issues sorted out. I don't think it is actually our problem. I documented it over in #89.

File: pysatModels/models/pydineof_dineof.py [hunk 3]
```diff
@@ -208,12 +206,19 @@ def download(date_array=None, tag=None, inst_id=None, data_path=None, **kwargs):
...
+                if not os.path.isfile(saved_local_fname):
+                    with open(saved_local_fname, 'wb') as open_f:
+                        open_f.write(req.content)
+            else:
+                warnings.warn('Unable to find remote file: {:}'.format(
+                    remote_path))
     else:
+        saved_local_fname = os.path.join(data_path, 'empty_file.txt')
+        with open(saved_local_fname, 'w') as open_f:
+            open_f.write(' ')
```
Comment (jklenzing) [inline pysatModels/models/pydineof_dineof.py:220]: Why do we create a file if there's nothing to download?

Comment (rstoneback) [reply pysatModels/models/pydineof_dineof.py:220]: Good question. 🐵 

It has been awhile but they may be some leftover test code intended to run on my local system... If so, then I must've forgot about it, and didn't see it when I copied code over from the other branch. 🙈

File: pysatModels/models/pydineof_dineof.py [hunk 4]
```diff
@@ -94,7 +92,7 @@ def init(self):
 
 
 def clean(self):
-    """Method to return pydineof data cleaned to the specified level, unused
+    """Clean pydineof data to the specified level, unused
     """
```
Comment (aburrell) [inline pysatModels/models/pydineof_dineof.py:96]:
```suggestion
    """Clean pydineof data to the specified level, unused."""
```

**Comment by rstoneback** [comment]
This is going to be replaced with #92 which fixes #82.

**Pull request merged**